src/ILLink.Shared/SharedStrings.resx

```diff
@@ -141,4 +141,28 @@
   <data name="RequiresAssemblyFilesMessage" xml:space="preserve">
     <value>Using member '{0}' which has 'RequiresAssemblyFilesAttribute' can break functionality when embedded in a single-file app.{1}{2}</value>
   </data>
+  <data name="BaseRequiresMismatchMessage" xml:space="preserve">
+    <value>Base member '{1}' with '{0}' has a derived member '{2}' without '{0}'. Add the '{0}' to '{1}'.</value>
```

Comment:
If we want to keep one error code and just have a bunch of different messages, we could have something like
```suggestion
  <data name="BaseRequiresMismatchMessage" xml:space="preserve">
    <value>{0}</value>
   </data>
```

and then when we create a diagnostic we would manually pull out the messages below and call

```C#
CreateDiagnostic(RequiresMismatchMessage, Location, string.Format(SharedStrings.BaseMismatchRequiresMessage, ..., ..., ...);
```